packages/asgardex-binance/src/client.ts

```diff
@@ -155,158 +173,253 @@ class Client implements BinanceClient {
+
+      return balances.map(balance => {
+        return {
+          asset: assetFromString(balance.symbol) || AssetBNB,
+          amount: assetToBase(assetAmount(balance.free, 8)),
+          frozenAmount: assetToBase(assetAmount(balance.frozen, 8)),
+        }
+      }).filter(balance => !asset || balance.asset === asset)
+    } catch (error) {
+      return Promise.reject(error)
+    }
   }
 
-  getTransactions = async (params: GetTxsParams = {}): Promise<TxPage> => {
-    const {
-      address = this.getAddress(),
-      blockHeight,
-      endTime,
-      limit,
-      offset,
-      side,
-      startTime,
-      txAsset,
-      txType,
-    } = params
+  getTransactions = async (params?: TxHistoryParams): Promise<TxsPage> => {
+    await this.bncClient.initChain()
 
     const clientUrl = `${this.getClientUrl()}/api/v1/transactions`
     const url = new URL(clientUrl)
-    if (address) url.searchParams.set('address', address)
-    if (blockHeight) url.searchParams.set('blockHeight', blockHeight.toString())
-    if (endTime) url.searchParams.set('endTime', endTime.toString())
-    if (limit) url.searchParams.set('limit', limit.toString())
-    if (offset) url.searchParams.set('offset', offset.toString())
-    if (side) url.searchParams.set('side', side.toString())
-    if (startTime) url.searchParams.set('startTime', startTime.toString())
-    if (txAsset) url.searchParams.set('txAsset', txAsset.toString())
-    if (txType) url.searchParams.set('txType', txType.toString())
 
-    await this.bncClient.initChain()
+    url.searchParams.set('address', params ? params.address : this.getAddress())
+    if (params && params.limit) {
+      url.searchParams.set('limit', params.limit.toString())
+    }
+    if (params && params.offset) {
+      url.searchParams.set('offset', params.offset.toString())
+    }
+    if (params && params.startTime) {
+      url.searchParams.set('startTime', params.startTime.toString())
+    }
 
     try {
-      const response = await axios.get<TxPage>(url.toString())
-      return response.data
+      const txHistory = await axios.get<BinanceTxPage>(url.toString()).then(response => response.data)
+      return {
+        total: txHistory.total,
+        txs: txHistory.tx.map(tx => {
+          const asset = assetFromString(`${AssetBNB.chain}.${tx.txAsset}`);
+
+          if (!asset) return null
+
+          return {
+            asset,
+            from: [
+              {
+                from: tx.fromAddr,
+                amount: assetToBase(assetAmount(tx.value, 8)),
+              }
+            ],
+            to: [
+              {
+                to: tx.toAddr,
+                amount: assetToBase(assetAmount(tx.value, 8)),
+              }
+            ],
+            date: new Date(tx.timeStamp),
+            type: getTxType(tx.txType),
+            hash: tx.txHash,
+          }
+        })
+        .filter(tx => tx).map(tx => tx!)
+      }
```

Comment:
[suggestion] Try `txHistory.tx.reduce(...)` instead of `txHistory.tx.map(...).filter(...).map(...)`